### PR TITLE
[RFC] Merge the CSS classes and remove the "space" setting

### PR DESCRIPTION
**Pull request by leofeyer**
This is a PR for https://github.com/contao/core/issues/6011 (and probably a dozen more). It merges the CSS classes if an element is included in another element and also removes the "space before/after" field.

### CODE REVIEW

**Comment by Toflar** [comment]
I think you should call a separate `getCssClassesAndId()` (or so) which can be overwritten in the child classes.

**Comment by leofeyer** [comment]
You can overwrite the class properties in the `generate()` method of the child class.

**Comment by Toflar** [comment]
I know. It's just a bit tedious because everybody has to do the `trim()` stuff and I thought now that we're changing it, why not? :)

**Comment by leofeyer** [comment]
If you override the parent method, don't you have to apply `trim()` as well?

**Comment by Toflar** [comment]
If you do

```php
/**
 * @return string
 */
public function getCssId();

/**
 * @return array
 */
public function getCssClasses();
```

not :-) Just an idea :)

**Comment by leofeyer** [comment]
I don't understand why a method saves us from using `trim()`. If I wanted to add a CSS class, here's what I'd do:

```php
public function getCssClasses()
{
    return trim(parent::getCssClasses() . ' custom-class');
}
```

Since I don't know if there is a parent CSS class, I **always** have to use `trim()`, don't I?

File: UPGRADE.md [hunk 1]
```diff
@@ -4,6 +4,20 @@ API changes
 Version 3.* to 4.0
 ------------------
 
+### Space before/after
+
+The field "space before/after" has been removed. Use a CSS class instead and
+define the spacing in your style sheet.
+
+
+### CSS classes of incuded elements
```
Comment (xchs) [inline UPGRADE.md]: Typo: "incuded" -> "included"

Comment (leofeyer) [inline UPGRADE.md]: Thanks @xchs, fixed in 26128ea71e8337493dec02dc45ac811638c2926b.

**Comment by leofeyer** [comment]
Merged in d2b1fa8663099c0060678c53d6ae8718175a566d.

**Pull request merged**